### DISCUSSION TITLE
how to force a DataTable to redraw the visible part of itself

### DISCUSSION THREAD

**Discussion by leemthai** [Textualize/textual#4865] [Q&A]
Sometimes I regenerate the rows and columns in a DataTable but the visible part of the table does not refresh properly. The label column is missing.

As a user, to 'fix' it I can either move the mouse left or right a bit (and the datatable instantly redraws properly) or use the mouse scrollwheel to scroll down and get a 'double view' of both:
a) the part that didn't redraw correctly
b)  the part that was initially off-screen

![Screenshot_20240810_114925](https://github.com/user-attachments/assets/9928607a-b021-40ac-8b4b-36bf50ab072a)

After regenerating rows and columns for a DataTable is there a call I make to ensure the visible part is repainted ?

Somehow the initial repaint is happening before the DataTable has finished being regenerated maybe? 

Any tips  or clues as to what is going wrong here?I'm pretty lost ...

**Comment by leemthai** [comment]
I think this shows it better, perhaps:
![Peek 2024-08-10 12-06](https://github.com/user-attachments/assets/fc1143d7-366a-4440-8d2f-7ede03f4c4fa)

**Comment by willmcgugan** [comment]
That isn't typical behavior of the DataTable. You would need to share code for us to understand what is going on. Ideally a short example that we can run here.

**Reply by leemthai** [reply]
I've actually had this "disappearing label column" quite a few times now. But when I rationalize my code it seems to magically disappear so I guess its various bugs in my stuff. I need to apply "KISS" more :)
